server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -148,6 +145,12 @@ protected AutoDateHistogramAggregationBuilder(AutoDateHistogramAggregationBuilde
         this.minimumIntervalExpression = clone.minimumIntervalExpression;
     }
 
+    @Override
+    protected ValuesSourceType defaultValueSourceType() {
+        // TODO: This should probably be DATE, but we're not failing tests with BYTES, so needs more tests?
```

Comment:
I'm aware of this TODO, I think it's okay to merge to master as is and fix there.